service/controller/resource/promremotewrite/delete.go

```diff
@@ -0,0 +1,62 @@
+package promremotewrite
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/v7/pkg/controller/context/resourcecanceledcontext"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (r *Resource) EnsureDeleted(ctx context.Context, obj interface{}) error {
+	r.logger.Debugf(ctx, "deleting")
+	{
+		//get remotewrite
+		remoteWrite, err := ToRemoteWrite(obj)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		selector, err := metav1.LabelSelectorAsSelector(&remoteWrite.Spec.ClusterSelector)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+		// fetch current prometheus
```

Comment:
As this code appears to be in both delete and create, what do you think about creating a function to fetch the prometheus based on a label selector ?